File: archive/crm_modules/Integraciones avanzadas2/integrations.py
import requests
from datetime import datetime

# Google Calendar API (ejemplo básico)
def add_event_to_google_calendar(event_title, start_time, end_time, calendar_id, token):
    url = f"https://www.googleapis.com/calendar/v3/calendars/{calendar_id}/events"
    headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}
    data = {
        "summary": event_title,
        "start": {"dateTime": start_time},
        "end": {"dateTime": end_time},
    }
    r = requests.post(url, json=data, headers=headers)
    return r.json()

# WhatsApp Business API
def send_whatsapp_message(phone_number, message, token):
    url = "https://graph.facebook.com/v17.0/YOUR_PHONE_NUMBER_ID/messages"
    headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}
    data = {
        "messaging_product": "whatsapp",
        "to": phone_number,
        "type": "text",
        "text": {"body": message}
    }
    r = requests.post(url, json=data, headers=headers)
    return r.json()

# Telefonía VoIP (Ejemplo con API ficticia)
def make_voip_call(from_number, to_number, token):
    url = "https://api.voipprovider.com/call"
    headers = {"Authorization": f"Bearer {token}"}
    data = {"from": from_number, "to": to_number}
    r = requests.post(url, json=data, headers=headers)
    return r.json()

File: archive/crm_modules/Integraciones-avanzadas/server/integrations.js
// server/integrations.js
const { google } = require('googleapis');
const axios = require('axios');

// --- Google Calendar ---
async function addEventToGoogleCalendar(auth, event) {
  const calendar = google.calendar({ version: 'v3', auth });
  return await calendar.events.insert({ calendarId: 'primary', requestBody: event });
}

// --- WhatsApp Business API (Ejemplo Twilio) ---
async function sendWhatsAppMessage(to, message) {
  const accountSid = process.env.TWILIO_SID;
  const authToken = process.env.TWILIO_TOKEN;
  const client = require('twilio')(accountSid, authToken);
  return await client.messages.create({
    from: 'whatsapp:' + process.env.TWILIO_WHATSAPP_FROM,
    to: 'whatsapp:' + to,
    body: message
  });
}

// --- Telefonía VoIP (Ejemplo Twilio) ---
async function makeCall(to, from, url) {
  const client = require('twilio')(process.env.TWILIO_SID, process.env.TWILIO_TOKEN);
  return await client.calls.create({ to, from, url });
}

module.exports = { addEventToGoogleCalendar, sendWhatsAppMessage, makeCall };
